jobs/resource_job.go

```diff
@@ -472,8 +472,20 @@ func (a JobsAPI) ListByName(name string, expandTasks bool) ([]Job, error) {
 }
 
 // List all jobs
-func (a JobsAPI) List() (l []Job, err error) {
-	l, err = a.ListByName("", false)
+func (a JobsAPI) List(nameFilter string) (l []Job, err error) {
+	if nameFilter == "" {
+		return a.ListByName("", false)
+	}
+
+	allJobs, err := a.ListByName("", false)
+	if err != nil {
+		return l, err
+	}
+	for _, job := range allJobs {
+		if strings.Contains(job.Settings.Name, nameFilter) {
+			l = append(l, job)
+		}
+	}
```

Comment:
I wouldn't put this into the API, but do the filtering in the data source itself.